examples/with-tailwind/packages/ui/tsup.config.ts

```diff
@@ -1,13 +1,46 @@
 import { defineConfig, Options } from "tsup";
+import path from "path";
+import fs from "fs-extra";
+
+const jsFileScan = (dir:string):string[] => {
+  
+  const paths = []
+  const nodes = fs.readdirSync(dir);
+  for (const node of nodes) {
+    const nodePath = path.join(dir, node);
+    const stat = fs.statSync(nodePath);
+    if (stat.isDirectory()) {
+      paths.push(...jsFileScan(nodePath));
+    } else if (stat.isFile() && (node.endsWith(".js") || node.endsWith(".mjs"))) {
+      paths.push(nodePath);
+    }
+  }
+
+  return paths;
+}
 
 export default defineConfig((options: Options) => ({
   treeshake: true,
   splitting: true,
   entry: ["src/**/*.tsx"],
-  format: ["esm"],
+  outDir: 'dist',
+  format: ['cjs'],
   dts: true,
   minify: true,
   clean: true,
```

Comment:
This `clean: true` causes VSCode to show missing types when restarting the dev server. I'm not sure how to mitigate this since removing this would make it so you could have outdated modules laying around while developing. 🤔

Reply:
I was looking into this too. I think having `clean` enabled deteriorates DX, sometime to the point of requiring a full manual browser refresh due to "missing" chunks. It was enabled before but I think we should try disabling it.

Reply:
turned clean to false in https://github.com/vercel/turbo/pull/6307/commits/13ab401a5d07ac41981b5f1a3aec307d6caf19a4,

much better local dev experience. no more nagging errors flashing up.

Reply:
added `rimraf` to clean up `dist` before build / dev commands in https://github.com/vercel/turbo/pull/6307/commits/ce24402d8b269121f0e32e49da37fd967a200049